catchup/service.go

```diff
@@ -710,6 +731,28 @@ func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.Asy
 				return
 			default:
 			}
+			if err == errNoBlockForRound {
```

Comment:
Are we not triggering this case in a test?

Reply:
it is really hard to write a deterministic test since most of the state is local to this function. So opt in not to write a test.